cloudevents/extensions/opcua.md

```diff
@@ -0,0 +1,335 @@
+# OPC UA
+
+This extension defines the mapping of [OPC UA](https://reference.opcfoundation.org/Core/Part1/v105/docs/)
+[PubSub](https://reference.opcfoundation.org/Core/Part14/v105/docs/) dataset to
+CloudEvents to allow seamless routing of OPC UA dataset messages via different
+protocols, it therefore provides a recommendation to map known REQUIRED and
+OPTIONAL attributes using other extensions as well as defines its own extension
+attributes.
+
+## Notational Conventions
+
+As with the main [CloudEvents specification](../spec.md), the key words "MUST",
+"MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+However, the scope of these key words is limited to when this extension is used.
+For example, an attribute being marked as "REQUIRED" does not mean it needs to
+be in all CloudEvents, rather it needs to be included only when this extension
+is being used.
+
+## Mapping of REQUIRED Attributes
+
+### id
+
+MUST map to [Network Message
+Header](https://reference.opcfoundation.org/Core/Part14/v105/docs/7.2.5.3#Table163)
+field `MessageId`.
+
+### source
+
+MUST either map to [Application
+Description](https://reference.opcfoundation.org/Core/Part4/v104/docs/7.1) field
+`applicationUri` of the OPC UA server or to a customer configured identifier
```

Comment:
Will a OPC UA expert know which to use in all cases or do we need to provide guidance?

Reply:
As the identifier are taken from the specification and the correct parts of the specifications are always linked, that should be fine for a OPC UA expert.